File: app/src/main/java/ru/serjik/engine/gl/BufferAllocator.java
package ru.serjik.engine.gl;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.nio.FloatBuffer;

public class BufferAllocator {
    public static FloatBuffer createFloatBuffer(int floatCount) {
        ByteBuffer byteBuffer = ByteBuffer.allocateDirect(floatCount * 4);
        byteBuffer.order(ByteOrder.nativeOrder());
        FloatBuffer floatBuffer = byteBuffer.asFloatBuffer();
        floatBuffer.position(0);
        return floatBuffer;
    }
}

File: app/src/main/java/ru/serjik/engine/gl/BufferAllocator.kt
package ru.serjik.engine.gl

import java.nio.ByteBuffer
import java.nio.ByteOrder
import java.nio.FloatBuffer

object BufferAllocator {
    fun createFloatBuffer(floatCount: Int): FloatBuffer {
        return ByteBuffer.allocateDirect(floatCount * 4).apply {
            order(ByteOrder.nativeOrder())
        }.asFloatBuffer().apply {
            position(0)
        }
    }
}
